mobify/platform-scaffold
# Add fallback fetch for service-worker-loader

In SFCC, downloading the `service-worker-loader` is only possible _without_ the `.js` extension. Thus, we attempt to first get the file with the extension, but if that fails, we retry without the extension.

**JIRA**: (link to JIRA ticket)

## Changes
- If fetching `service-worker-loader.js` fails, then try fetching it without the `.js` extension

## Todos
- ~~(if applicable) analytics events instrumented to measure impact of change~~
- ~~Add a high-level description of your changes to the CHANGELOG.md, including a link to the PR with your changes~~

## How to test-drive this PR
- In `web/app/main.jsx`, comment out the Merlin's `initConnector` import, and uncomment the SFCC import. It should look like...
    ```jsx
    // import initConnector from './init-merlins-connector'
    import initConnector from './init-sfcc-connector'
    ```
- Run `npm start`
- Preview [SFCC Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- After the page loads, verify that the `service-worker-loader` file downloaded correctly. **Note** that you may see `service-worker-loader.js` (the one with the extension) will have failed, but the network request without the extension should succeed.
